### ISSUE TITLE
Error filtering in aws-sdk-transcribestreamingservice overly restrictive

### ISSUE THREAD

**Issue by TJNII** [aws/aws-sdk-ruby#2082]
### Issue description

When using the aws-sdk-transcribestreamingservice gem I noticed exceptions and log messages have important information redacted out of them.

The following was the exception text when too large of a block size was used:

```
{:message=>"[FILTERED]", :event_type=>:bad_request_exception}
```

By enabling http debugging I was able to see that :message should have shown the following, non-sensitive information:

```
{\"Message\":\"Your stream is too big. Reduce the frame size and try your request again.\"}
```

Furthermore when printing events to get a better sense of what events are available I noticed the event types were also being redacted:

```
{:transcript=>{:results=>[{:result_id=>"02856682-bcd0-4cf5-9d30-c1354d7ed447", :start_time=>0.12, :end_time=>8.15, :is_partial=>false, :alternatives=>[{:transcript=>"Inscription. Test. One two Transcription Test. One two. Testing testing one two three four Testing testing one two, three four.", :items=>[{:start_time=>0.14, :end_time=>0.58, :type=>"[FILTERED]", :content=>"Inscription"}, {:start_time=>0.58, :end_time=>0.58, :type=>"[FILTERED]", :content=>"."}, {:start_time=>0.59, :end_time=>0.85, :type=>"[FILTERED]", :content=>"Test"}, {:start_time=>0.85, :end_time=>0.85, :type=>"[FILTERED]", :content=>"."}, {:start_time=>0.86, :end_time=>1.01, :type=>"[FILTERED]", :content=>"One"}, {:start_time=>1.02, :end_time=>1.48, :type=>"[FILTERED]", :content=>"two"}, {:start_time=>1.5, :end_time=>2.13, :type=>"[FILTERED]", :content=>"Transcription"}, {:start_time=>2.14, :end_time=>2.39, :type=>"[FILTERED]", :content=>"Test"}, {:start_time=>2.39, :end_time=>2.39, :type=>"[FILTERED]", :content=>"."}, {:start_time=>2.4, :end_time=>2.55, :type=>"[FILTERED]", :content=>"One"}, {:start_time=>2.56, :end_time=>2.87, :type=>"[FILTERED]", :content=>"two"}, {:start_time=>2.87, :end_time=>2.87, :type=>"[FILTERED]", :content=>"."}, {:start_time=>3.41, :end_time=>3.78, :type=>"[FILTERED]", :content=>"Testing"}, {:start_time=>3.79, :end_time=>4.34, :type=>"[FILTERED]", :content=>"testing"}, {:start_time=>4.44, :end_time=>4.65, :type=>"[FILTERED]", :content=>"one"}, {:start_time=>4.66, :end_time=>4.82, :type=>"[FILTERED]", :content=>"two"}, {:start_time=>4.83, :end_time=>5.04, :type=>"[FILTERED]", :content=>"three"}, {:start_time=>5.05, :end_time=>5.47, :type=>"[FILTERED]", :content=>"four"}, {:start_time=>5.94, :end_time=>6.29, :type=>"[FILTERED]", :content=>"Testing"}, {:start_time=>6.3, :end_time=>6.88, :type=>"[FILTERED]", :content=>"testing"}, {:start_time=>6.96, :end_time=>7.14, :type=>"[FILTERED]", :content=>"one"}, {:start_time=>7.15, :end_time=>7.31, :type=>"[FILTERED]", :content=>"two"}, {:start_time=>7.31, :end_time=>7.31, :type=>"[FILTERED]", :content=>","}, {:start_time=>7.32, :end_time=>7.51, :type=>"[FILTERED]", :content=>"three"}, {:start_time=>7.52, :end_time=>7.83, :type=>"[FILTERED]", :content=>"four"}, {:start_time=>7.83, :end_time=>7.83, :type=>"[FILTERED]", :content=>"."}]}]}]}, :event_type=>:transcript_event}
```

This seems like an undesired side-effect, and a quick search didn't yield how to turn this filtering off.

### Gem name ('aws-sdk', 'aws-sdk-resources' or service gems like 'aws-sdk-s3') and its version

```
GEM
  remote: https://rubygems.org/
  specs:
    aws-eventstream (1.0.3)
    aws-partitions (1.182.0)
    aws-sdk-core (3.57.0)
      aws-eventstream (~> 1.0, >= 1.0.2)
      aws-partitions (~> 1.0)
      aws-sigv4 (~> 1.1)
      jmespath (~> 1.0)
    aws-sdk-transcribestreamingservice (1.6.0)
      aws-sdk-core (~> 3, >= 3.56.0)
      aws-sigv4 (~> 1.1)
    aws-sigv4 (1.1.0)
      aws-eventstream (~> 1.0, >= 1.0.2)
    http-2 (0.10.1)
    jmespath (1.4.0)
```

### Version of Ruby, OS environment

ruby 2.6.3p62 (2019-04-16 revision 67580) [x86_64-linux]

### Code snippets / steps to reproduce

Add the following event handler to the demo app at https://aws.amazon.com/blogs/developer/announcing-amazon-transcribe-streaming-transcription-support-in-the-aws-sdk-for-ruby/ to see the event filtering.

```
logger = Logger.new(STDOUT)
output_stream.on_event do |event|
  logger.debug("Event: #{event}")
end
```

**Comment by cjyclaire** [comment]
Appreciate the report! yeah, this is a side affect that  sensitive [`filter`](https://github.com/aws/aws-sdk-ruby/blob/master/gems/aws-sdk-core/lib/aws-sdk-core/log/param_filter.rb#L14) applied to all services, ideally, SDK should apply those filters by services, or at least, allow more customizable disabling logic per client (or for some params)

**Comment by mullermp** [comment]
Hi @TJNII this has been addressed in #2342. Sorry for the delay!
